dumpling/export/dump.go

```diff
@@ -1337,6 +1339,26 @@ func resolveAutoConsistency(d *Dumper) error {
 	default:
 		conf.Consistency = ConsistencyTypeNone
 	}
+
+	if conf.Consistency == ConsistencyTypeFlush {
+		grants, err := dbutil.ShowGrants(d.tctx.Context, d.dbHandle, "", "")
+		if err != nil {
+			d.tctx.L().Error("failed to query SHOW GRANTS", zap.Error(err))
+			return nil
+		}
+		var privMap = map[pmysql.PrivilegeType]map[string]map[string]struct{}{}
+		privMap[pmysql.SuperPriv] = nil
```

Comment:
Do you mean we can check `ReloadPriv` for `ConsistencyTypeLock`?

Reply:
I mean we can check whether do users have `SuperPriv` or (`ReloadPriv` + `LOCK TABLES Priv` on all tables)

Reply:
I don't think FTWRL can be done with `RELOAD` and `LOCK TABLES`. Below is an output from Aurora, it has these two privilege but it should not support FTWRL

 https://github.com/pingcap/tiflow/issues/6448

Reply:
It sames that Aurora's behavior is a little different from mysql community version.
```
mysql> SHOW GRANTS for 'dm_incremental'@'%';
+----------------------------------------------------------+
| Grants for dm_incremental@%                              |
+----------------------------------------------------------+
| GRANT RELOAD ON *.* TO 'dm_incremental'@'%' |
+----------------------------------------------------------+
1 row in set (0.01 sec)

$ mysql -h 127.0.0.1 -P 3306 -u dm_incremental -p123456
mysql: [Warning] Using a password on the command line interface can be insecure.
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 13
Server version: 5.7.32 MySQL Community Server (GPL)

Copyright (c) 2000, 2022, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql>
mysql> flush tables with read lock;
Query OK, 0 rows affected (0.00 sec)
```

Reply:
maybe because SUPER is deprecated in MySQL 8.0. I think I should execute FTWRL here for a test.

https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_super